server/lib/SMS-sender.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const { SNSClient, PublishCommand } = require('@aws-sdk/client-sns');
+const config = require('./config');
+
+/**
+ * This is configured SNSClient that is used with AWS-SNS API.
+ * Contains an error if something went wrong during initialization.
+ * This constant is automatically initialized.
+ * @type {SNSClient}
+ */
+const client = function() {
+    try {
+        process.env.AWS_ACCESS_KEY_ID = config.SMS.keyID;
+        process.env.AWS_SECRET_ACCESS_KEY = config.SMS.secretKey;
+        return new SNSClient({region: config.SMS.region});
+    }
+    catch (error) {
+        return error;
+    }
+}();
+
+/**
+ * Sends SMS message via AWS-Simple Notification Service.
+ * @param {string} phoneNumber - The phone number of the recipient in E.164 format.
+ * @param {string} text - The text of the SMS, it is better to keep it short.
+ * @returns {Promise<*>} Status of the operation.
+ */
+async function sendSMS(phoneNumber, text) {
+    try {
+        const params = {PhoneNumber: phoneNumber, Message: text};
+        return await client.send(new PublishCommand(params));
+    }
+    catch (error) {
+        return error;
```

Comment:
Same, see above

Reply:
Yes, but you are not checking for errors anywhere, and you need to at least log that something has failed. Otherwise there will be no information about the failure.

https://github.com/JanJanda/ivis-core/blob/73a9ae10b5977a6b3f7a61a63bd4a0c62a7cdf67/server/models/alerts-log.js#L49-L52

Also I think it's better design to consciously either set it up and then you expect it to work, therefore letting it throw on error should be expected. This is a wrong place to gateway this if not set up properly. It should never get to this place. Otherwise the system is actually trying to send notification, but failing. It should never get to this point if not set up, or disabled.

Thank you for all the other changes!

Reply:
I removed silent exceptions, but left the check whether it is enabled. Logs of notifications should be completely independent of alert, because notifications may be used independently in the future. It means that we need new table in DB for that. I left that as a todo for future.

Reply:
Okay then, that's fine. Thank you!